### PR TITLE
Bump minimal supported Boost to 1.58.0

### PR DESCRIPTION
**Pull request by K-ballo**

### CODE REVIEW

File: docs/manual/build_system/prerequisites.qbk [hunk 1]
```diff
@@ -63,7 +63,7 @@ favorite compiler with __hpx__ visit __stellar_hpx_buildbot__.
     [[__cmake__                  ][3.9.0              ][3.3.2           ]]
 
     [[[*Required Libraries]]]
-    [[__boost_libraries__        ][1.57.0 or newer    ][1.55.0          ]]
+    [[__boost_libraries__        ][1.57.0 or newer    ][1.58.0          ]]
```
Comment (msimberg) [inline docs/manual/build_system/prerequisites.qbk]: Recommended version should presumably be at least as big as the minimum version?

File: docs/manual/build_system/prerequisites.qbk [hunk 2]
```diff
@@ -90,8 +90,7 @@ favorite compiler with __hpx__ visit __stellar_hpx_buildbot__.
     [[__cmake__                  ][3.9.0                ][3.3.2           ]]
 
     [[[*Required Libraries]]]
-    [[__boost__                  ][1.57.0 or newer    ][1.55.0          ]
-     [See below for an important limitation when using Boost V1.55.0.]]
+    [[__boost__                  ][1.57.0 or newer    ][1.58.0          ]]
```
Comment (msimberg) [inline docs/manual/build_system/prerequisites.qbk]: Same here.

**Comment by K-ballo** [comment]
Bumped recommended Boost version to 1.62.0

**Comment by hapoo** [comment]
cmake should complain about boost version as well. right now cmake accepts the older boost in configuration and then compile fails. http://rostam.cct.lsu.edu/builders/hpx_gcc_4_9_boost_1_56_centos_x86_64_debug/builds/298

**Pull request merged**